application/controllers/AuthenticationController.php

```diff
@@ -35,11 +39,34 @@ public function loginAction()
             $this->redirectNow(Url::fromPath('setup'));
         }
         $form = new LoginForm();
+
+        if (RememberMe::hasCookie()) {
+            $authenticated = false;
+
+            try {
+                $rememberMeOld = RememberMe::fromCookie();
+                $authenticated = $rememberMeOld->authenticate();
+                if ($authenticated) {
+                    $rememberMe = $rememberMeOld->renew();
+                    $this->getResponse()->setCookie($rememberMe->getCookie());
+                    $rememberMe->persist($rememberMeOld->getAesCrypt()->getIv());
+                }
+            } catch (RuntimeException $e) {
+                Logger::error("Can't authenticate user via remember me cookie: %s", $e->getMessage());
+            } catch (AuthenticationException $e) {
+                Logger::error($e);
+            }
+
+            if (! $authenticated) {
+                $this->getResponse()->setCookie(RememberMe::forget());
+            }
+        }
+
         if ($this->Auth()->isAuthenticated()) {
             // Call provided AuthenticationHook(s) when login action is called
             // but icinga web user is already authenticated
             AuthenticationHook::triggerLogin($this->Auth()->getUser());
-            $this->redirectNow($form->getRedirectUrl());
+            $this->redirectNow($this->params->get('redirect', $form->getRedirectUrl()));
```

Comment:
Why is this change necessary? That's being used by the form anyway, so why are you fetching the param directly again?

Reply:
Because `$form->getRedirectUrl()` redirects to the default page (dashboard).

Reply:
Not for me. Regardless of whether I tick "stay logged in" on or not.

Reply:
You can reproduce it as follows:

1. Log in with "stay logged" checked.
2. Go to a specific url, copy it and close the browser completely.
3. Open the browser and enter the copied URL.

If you are redirected only with $form->getRedirectUrl(), you will end up on dashbords.

(I think it's because if cookie is present, you won't be redirected to login page at all. So the hidden redirect element is not filled and you end up on default url ).

To change this behaviour, I have added this change.

Reply:
I see. Thanks for the explanation.